add atomicPhysicsCI setup

BrianMarre opened the pull request:
adds the atomic Physics test setup and validation scripts used for runtime tests.

- [x] needs to be rebased and updated to PR #5119
- [x] rebase against https://github.com/ComputationalRadiationPhysics/picongpu/pull/5134

BrianMarre committed:
add atomicPhysics example

adds the atomicPhysicsCI runtime test setup previously held in a separate repository to PIConGPU

BrianMarre commented:
@ikbuibui please add the cfg you are using and maybe a few notes regarding the CI workflow. Thanks!

ikbuibui committed:
Add CFGs and submitAction to atomicPhysics example

ikbuibui commented:
The way to use the atomic physics test can be simply to copy the test code outside the picongpu source, and execute 
``./validate/validateLocally.sh``

I have added the CFGs I am using for the JUBE CI scaling tests, in folders inside ``etc/`` corresponding to the tests. These CFGs are designed to run on NVIDIA A100 (40 GB) GPUs. The ``single/1.cfg`` is a single GPU test, for the other CFGs the number (eg. ``strong_tiny/4.cfg``) refers to the number of nodes.
One can execute these CFGs with TBG, but validation wont be performed. The user will have to use the custom ``submitAction.sh`` provided with the cfg, or copy the validation folder to the location of the results manually. Finally the user must execute ``validate.sh``. The user should be careful to set the correct CFG and ``submitAction`` paths. 

In the JUBE CI, the workflow is as follows
JUBE calls TBG for the scaling test we want to perform (weak/strong, different node sizes). Each scaling test has a set of CFG files associated. JUBE runs the CFGs one by one.
Using a slightly modified ``submitAction.sh``, the TBG call copies over the validation folder to where the results are generated.
In the TPL file, after the ``srun`` to start the simulation, we call the validation script and we are done.

ikbuibui commented:
Updated the PR to not hold all the scaling CFGs. Now there are two CFGs in `etc/`, written for use on nodes with 4 NVIDIA A100 (40 GB) GPUs. 
``1.cfg`` holds a single GPU config.
``4.cfg`` holds a 4 node, 4 GPUs per node config.

ikbuibui committed:
reduce number cfg-files in atomiPhysics example

psychocoderHPC commented:
Whats the difference to `tests/compileAtomicPhysics`. If we have an example than the atomic  physics code is already compile test and there is no explicit compile test required.

BrianMarre committed:
update atomicPhysics example README

BrianMarre committed:
remove old atomicPhysics compile test

ikbuibui committed:
disable force constant electron temperature by default

disables the atomicPhysics debug option force constant electron
temperature in the atomicPhysics example by default

To renenabled use the cmake flag preset 1

ikbuibui committed:
make cmakeFlags for CI test case executable in a CI

switches the check for enabling he cmake flags set 1

ci: picongpu

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+
+Any simulation using atomicPhysics(FLYonPIC) requires a user provided set of atomic input data files, these files are not provided with PIConGPU due to licensing issues.
+
+This simulation is intended to use the atomic input data files from `https://github.com/ComputationalRadiationPhysics/FLYonPICAtomicTestData`.
```
Should we link to a private repository?

BrianMarre replied:
Well so long as it is private the URL will not resolve unless the user has logged in and the user is authorized.
And it's the easiest way to distribute the input files to users.

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+Any simulation using atomicPhysics(FLYonPIC) requires a user provided set of atomic input data files, these files are not provided with PIConGPU due to licensing issues.
+
+This simulation is intended to use the atomic input data files from `https://github.com/ComputationalRadiationPhysics/FLYonPICAtomicTestData`.
+If you have access to this repository by `source ./etc/getAtomicInputData.sh` to copy a set of atomic input files to the `atomicInputData`-directory before running the simulation.
```
```suggestion
If you have access to this repository, execute `./etc/getAtomicInputData.sh` to copy a set of atomic input files to the `atomicInputData`-directory before running the simulation.
```

BrianMarre replied:
done

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+
+This setup models a homogeneous charge neutral copper plasma with an ion density of 2e28 1/m^3 and an initial copper charge state of 2.
+
+Copper ions are initialized cold, no initial momentum, in the ground state, while electrons are given an initial Maxwellian-Temperature of 1 keV.
```
```suggestion
Copper ions are initialized cold, with no initial momentum, in the ground state, while electrons are given an initial Maxwellian-Temperature of 1 keV.
```

BrianMarre replied:
done

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+
+From the root directory
+1.) run picongpu by hand using `./bin/picongpu -g 1 1 1 -d 1 1 1 -s 25 --periodic 1 1 1`
+2.) run `source ./validate/validateLocally.sh` to validate the simulation results
```
```suggestion
2.) run `./validate/validateLocally.sh` to validate the simulation results
```
Shouldn't source unless required

BrianMarre replied:
done

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+To build the run-time test use the following steps:
+1.) copy the atomic physics example using `pic-create` to a new setup folder
+2.) go inside the new setup folder
+2.) provide the atomic input data using `source ./etc/getAtomicInputData.sh`, requires access to the test setup.
```
```suggestion
2.) provide the atomic input data using `./etc/getAtomicInputData.sh`, requires access to the test setup.
```
Shouldn't source unless required

BrianMarre replied:
done

ikbuibui commented on share/picongpu/examples/AtomicPhysics/README.md:
```diff
@@ -0,0 +1,83 @@
+Run Using TBG
+_____________
+
+The setup comes with `.cfg`-files designed to run on NVIDIA A100 (40 GB) GPU nodes from the JUBE CI.
```
```suggestion
The setup comes with `.cfg`-files designed to run on NVIDIA A100 (40 GB) GPU nodes from JURECA using the JUBE CI.
```

BrianMarre replied:
done

ikbuibui reviewed (commented):
Just comments on the documentation

BrianMarre committed:
reviewer comments

ci: picongpu

psychocoderHPC commented on share/picongpu/examples/AtomicPhysics/include/picongpu/param/speciesDefinition.param:
```diff
@@ -32,6 +32,7 @@
 
 #pragma once
 
+#include "picongpu/simulation_defines.hpp"
```
this wile dows not exits anymore and it would be also wrong to include this within a param file

BrianMarre committed:
remove reference to simulation defines from atomicPhysics example

ci: picongu
